nilearn/plotting/surf_plotting.py

```diff
@@ -341,12 +341,12 @@ def _get_ticks_matplotlib(vmin, vmax, cbar_tick_format):
     n_ticks = 5
     # ...unless we are dealing with integers with a small range
     # in this case, we reduce the number of ticks
-    if cbar_tick_format == "%i" and vmax - vmin < n_ticks:
+    if cbar_tick_format == "%i" and vmax - vmin < n_ticks - 1:
         ticks = np.arange(vmin, vmax + 1)
-        n_ticks = len(ticks)
     else:
         ticks = np.linspace(vmin, vmax, n_ticks)
-    return ticks
+    # remove duplicate ticks when vmin == vmax
+    return np.array(list(dict.fromkeys(ticks)))
```

Comment:
Duplicates can only ever happen when using `np.linspace` so I'd only use `np.unique` (or `np.array(list(dist.fromkeys(...` if you prefer) with that function.

Reply:
Note that I'm only suggesting this for the reader so that they don't wonder how duplicates might happen when using `np.arange`. I'm not suggesting this for performance reasons because the array is so small that it doesn't really make a difference ;)